pxr/usd/usd/coreValidators.cpp

```diff
@@ -12,26 +12,44 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
-TF_REGISTRY_FUNCTION(UsdValidationRegistry)
+static
+UsdValidationErrorVector
+_GetCompositionErrors(const UsdStagePtr &usdStage)
 {
-    const UsdValidateStageTaskFn stageTaskFn = 
-        [](const UsdStagePtr &usdStage) {
-            UsdValidationErrorVector errors;
-            const PcpErrorVector pcpErrors = usdStage->GetCompositionErrors();
-            errors.reserve(pcpErrors.size());
-            for (const PcpErrorBasePtr &pcpError : pcpErrors) {
-                UsdValidationErrorSites errorSites = {
-                    UsdValidationErrorSite(usdStage, pcpError->rootSite.path)};
-                errors.emplace_back(UsdValidationErrorType::Error, 
-                                    std::move(errorSites), 
-                                    pcpError->ToString());
-            }
-            return errors;
-        };
+    UsdValidationErrorVector errors;
+    const PcpErrorVector pcpErrors = usdStage->GetCompositionErrors();
+    errors.reserve(pcpErrors.size());
+    for (const PcpErrorBasePtr &pcpError : pcpErrors) {
+        UsdValidationErrorSites errorSites = {
+                UsdValidationErrorSite(usdStage, pcpError->rootSite.path)};
+        errors.emplace_back(UsdValidationErrorType::Error,
+                            std::move(errorSites),
+                            pcpError->ToString());
+    }
+    return errors;
+}
 
+static
+UsdValidationErrorVector
+_GetStageMetadataErrors(const UsdStagePtr &usdStage)
+{
+    UsdValidationErrorVector errors;
+    if (!usdStage->GetDefaultPrim()) {
+        errors.emplace_back(UsdValidationErrorType::Error,
+                            UsdValidationErrorSites{UsdValidationErrorSite(usdStage, SdfPath("/"))},
+                            "Stage has missing or invalid defaultPrim.");
```

Comment:
Maybe include the path to the rootLayer here in the reported error string.

Reply:
Updated